### DISCUSSION TITLE
[Feature request] 废弃`@midwayjs/decorator`

### DISCUSSION THREAD

**Discussion by zanminkian** [midwayjs/midway#3846] [Ideas]
### The feature, motivation（功能、动机）

全是从 `@midwayjs/core` 重新导出，没看出必要性。用户的项目里面这两个包混用，这个包只会让项目代码更混乱。

**Comment by czy88840616** [comment]
脚手架和文档应该都已经移除了 decorator，所有的官方组件中 decorator 的依赖也早就移除了，考虑到业务有依赖，不强制处理。

从 v4 起不再提供。

**Reply by waitingsong** [reply]
v4 准备用 js 新的装饰器方案么

**Reply by czy88840616** [reply]
目前看用不了，参数装饰器没有，metadata也没实现
